test/parallel/test-fs-watch-file-nonexistent.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var fs = require('fs');
+
+var filename = '/path/to/file/that/does/not/exist';
+
+fs.watchFile(filename, function() {
+  throw new Error('callback should not be called for non-existent files');
```

Comment:
@thefourtheye pointed out in an earlier revision that this means that the callback won't get called on ENOENT.

I think that already happens if you watch e.g. `/etc/shadow` because that's going to fail with EPERM or -1, passing the `newStatus === -1` check in `fs.watchFile()`.

That does seem unfortunate, though.  Invoking the callback at least gives the user a chance to handle the ENOENT case because then you can check that `curr.ino > 0`.